### PR TITLE
RxJS upgrade to v6, plus a few lint corrections.

### PR DESCRIPTION
**Pull request by nazrodriguez**
Luckily a headache-free, easy migration to an already very popular new version of RxJS. I hope we can agree on that! Regards.

### CODE REVIEW

**Comment by kaichancoder** [comment]
Hi, just wondering when this will be merged? Looking forward to v6. Thanks!

**Comment by SamuelMarks** [comment]
In the meantime @nazrodriguez can you throw in a build to your releases?

Tried:
```
npm i -S https://api.github.com/repos/nazrodriguez/ng2-completer/tarball/master
```

But getting these errors:
> Can't resolve 'ng2-completer'

---

Thanks

**Comment by nazrodriguez** [comment]
@SamuelMarks build uploaded!

Cheers,
Naza.

**Comment by oscarrodar** [comment]
@oferh when will this PR be merged?

**Comment by Danieltamir** [comment]
@oferh +1 to @oscarrodar

**Comment by sar13** [comment]
How can i use this before this gets pulled into the branch?

**Comment by stsishkou** [comment]
Dear author, could you merge such branch, in case it is not possible also let know all component consumers

File: package.json [hunk 1]
```diff
@@ -99,5 +99,8 @@
     "webpack-dev-server": "^2.9.4",
     "webpack-merge": "^4.1.0",
     "zone.js": "0.8.18"
+  },
+  "dependencies": {
```
Comment (stsishkou) [inline package.json]: Do 'compat' really needed?

Comment (nazrodriguez) [reply package.json]: For this first merge to go smoothly, I’d leave it. We can see to suppressing it if all works without it.

**Pull request merged**